### PR TITLE
KSQL-2630: Update Windows section with out-of-order-record and grace-period conc…

### PR DESCRIPTION
**Pull request by JimGalasyn**
This PR cribs updated content from @londoncalling's Kafka Streams updates, [PR #1993 DOCS-1429 Grace Period vs. Retention Time](https://github.com/confluentinc/docs/pull/1993) and [PR #2018 grace vs retention, clarify out-of-order](https://github.com/confluentinc/docs/pull/2018/).

### CODE REVIEW

**Comment by londoncalling** [comment]
@JimGalasyn shall I put all this in an include and we can both use it?

**Comment by JimGalasyn** [comment]
@londoncalling hmm, that's an excellent idea -- I'll say yes!

**Comment by londoncalling** [comment]
@JimGalasyn, cool, I'll add the includes now, and ping you when done.

**Comment by JimGalasyn** [comment]
@londoncalling Sweet! I'll close this PR and post a new one with your include file.

File: docs/concepts/time-and-windows-in-ksql-queries.rst [hunk 1]
```diff
@@ -180,36 +187,26 @@ windows per record key.
 
 When using windows in your KSQL queries, aggregate functions are applied only
 to the records that occur within a specific time window. Records that arrive
-late are handled as you might expect: although the time window they belong to
-has expired, the late records are still associated with the correct window.
+out-of-order are handled as you might expect: although the time window they belong to
+has expired, the out-of-order records are still associated with the correct window.
```
Comment (big-andy-coates) [inline docs/concepts/time-and-windows-in-ksql-queries.rst]: I had to read this a few times to try to understand it and I'm still not sure I do.

What do you mean by 'has expired' here?  How, if at all, does expiry relate to the 'grace period' you mention below?

Do you mean `expired` as in the normal progression of timestamps in the stream of messages has moved beyond the end of the window?

File: docs/concepts/time-and-windows-in-ksql-queries.rst [hunk 2]
```diff
@@ -180,36 +187,26 @@ windows per record key.
...
-late are handled as you might expect: although the time window they belong to
-has expired, the late records are still associated with the correct window.
+out-of-order are handled as you might expect: although the time window they belong to
+has expired, the out-of-order records are still associated with the correct window.
 
-You can specify a retention period for the window in your KSQL queries. This
-retention period controls how long KSQL waits for out-of-order or late-arriving
-records for a given window. If a record arrives after the retention period of a
-window has passed, the record is discarded and isn't processed in that window.
+You can specify a *grace period* for the window in your KSQL queries. This
```
Comment (big-andy-coates) [inline docs/concepts/time-and-windows-in-ksql-queries.rst]: How do you specify the 'grace period'?  Is there an example in `grace-period-late-events.rst`?  Could you add it into the review to help my understanding?

Comment (londoncalling) [reply docs/concepts/time-and-windows-in-ksql-queries.rst]: @mjsax , @big-andy-coates 's questions are for the KSQL docs that incorporate content from Streams docs related to "grace period". @JimGalasyn and I used `includes` to share content directly on the tip ("grace period vs. retention time") and the paragraph above it starting with "Out-of-order records are always possible in the real world ..", but the rest of the content we had to each write separately to be specific to KSQL or Streams.

@big-andy-coates 's questions are on the KSQL specific content. Can you review and clarify for Andy,  perhaps provide an example, and/or let us know how we should change what we have here?

Comment (mjsax) [reply docs/concepts/time-and-windows-in-ksql-queries.rst]: Not sure if KSQL exposed "grace period" to the user already? It was added to Kafka Streams in 2.1 release, but maybe there is no equivalent in KSQL yet -- KSQL also does not yet support `suppress()` operator.

File: docs/concepts/time-and-windows-in-ksql-queries.rst (hunk 1 above)
Comment (mjsax) [reply docs/concepts/time-and-windows-in-ksql-queries.rst]: I agree with Andi, that "has expired" might not be the best terminology. What about `although the window end time passed,` ?

File: docs/concepts/time-and-windows-in-ksql-queries.rst (hunk 2 above)
Comment (londoncalling) [reply docs/concepts/time-and-windows-in-ksql-queries.rst]: @JimGalasyn, see ^^ @mjsax comment, maybe explore whether this is exposed yet? If not, we can re-introduce here when it is ..?

Comment (JimGalasyn) [reply docs/concepts/time-and-windows-in-ksql-queries.rst]: @mjsax @big-andy-coates @londoncalling  Good feedback, I'll post another PR in a minute.

**Comment by JimGalasyn** [comment]
Continued in #2921.

**Pull request merged**